src/main/java/org/sonarlint/intellij/ui/CurrentFilePanel.java

```diff
@@ -120,49 +121,37 @@ public void update(@Nullable VirtualFile file, @Nullable Collection<LiveIssue> i
-      statusText.setText(emptyText);
-      if (liveIssues == null && (analyzeFilesAction.getTemplateText() != null)) {
-        statusText.appendLine(analyzeFilesAction.getTemplateText(), SimpleTextAttributes.LINK_PLAIN_ATTRIBUTES,
+    if (file == null) {
+      statusText.setText("No file opened in the editor");
+      disableEmptyDisplay(false);
+      treeBuilder.updateModel(Map.of());
+      oldTreeBuilder.updateModel(Map.of());
+      return;
+    }
+    if (issues == null) {
+      statusText.setText("No analysis done on the current opened file");
+      var templateText = analyzeFilesAction.getTemplateText();
+      if (templateText != null) {
+        statusText.appendLine(templateText, SimpleTextAttributes.LINK_PLAIN_ATTRIBUTES,
           ignore -> ActionUtil.invokeAction(analyzeFilesAction, this, CurrentFilePanel.SONARLINT_TOOLWINDOW_ID, null, null));
         statusText.appendText(" (Ctrl+Shift+S)");
       }
-    } else {
-      emptyText = "No file opened in the editor";
-      statusText.setText(emptyText);
-    }
-
-    if (liveIssues == null) {
-      liveIssues = Collections.emptyList();
+      issues = Collections.emptyList();
     }
-    update(file, List.copyOf(liveIssues), emptyText);
-  }
 
-  private void update(@Nullable VirtualFile file, Collection<LiveIssue> issues, String emptyText) {
-    if (file == null) {
-      disableEmptyDisplay(false);
-      treeBuilder.updateModel(Map.of(), emptyText);
-      oldTreeBuilder.updateModel(Map.of(), emptyText);
+    this.currentFile = file;
+    this.currentIssues = issues;
+    disableEmptyDisplay(!issues.isEmpty());
+    if (getService(project, CleanAsYouCodeService.class).shouldFocusOnNewCode()) {
+      var oldIssues = issues.stream().filter(not(LiveFinding::isOnNewCode)).collect(Collectors.toList());
+      var newIssues = issues.stream().filter(LiveFinding::isOnNewCode).collect(Collectors.toList());
+      treeBuilder.updateModel(Map.of(file, newIssues));
+      oldTreeBuilder.updateModel(Map.of(file, oldIssues));
+      oldTree.setVisible(true);
     } else {
-      this.currentFile = file;
-      this.currentIssues = issues;
-      disableEmptyDisplay(!issues.isEmpty());
-      if (getGlobalSettings().isFocusOnNewCode()) {
-        var oldIssues = issues.stream().filter(not(LiveFinding::isOnNewCode)).collect(Collectors.toList());
-        var newIssues = issues.stream().filter(LiveFinding::isOnNewCode).collect(Collectors.toList());
-        treeBuilder.updateModel(Map.of(file, newIssues), emptyText);
-        oldTreeBuilder.updateModel(Map.of(file, oldIssues), "No older issues");
-        oldTree.setVisible(true);
-      } else {
-        treeBuilder.updateModel(Map.of(file, issues), emptyText);
-        oldTreeBuilder.updateModel(Collections.emptyMap(), emptyText);
-        oldTree.setVisible(false);
-      }
+      treeBuilder.updateModel(Map.of(file, issues));
```

Comment:
We can extract lines 152 153 and 154 as a method to avoid duplication with lines 148 149 150

Reply:
I suggest to rework the code a bit later